contracts/mainnet/connectors/morpho-compound/main.sol

```diff
@@ -0,0 +1,440 @@
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogBorrowWithMaxGas(address,address,uint256,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_maxGasForMatching,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Withdraw ETH/ERC20_Token.
+	 * @notice Withdraw a token from Morpho Compound.
+	 * @param _tokenAddress The address of underlying token to withdraw.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to withdraw.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to withdraw. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens withdrawed.
+	 */
+	function withdraw(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+		bool _isETH = _tokenAddress == ethAddr;
+		address _token = _isETH ? wethAddr : _tokenAddress;
+
+		if (_amt == uint256(-1)) {
+			(, , _amt) = morphoCompoundLens.getCurrentSupplyBalanceInOf(
+				_poolTokenAddress,
+				address(this)
+			);
+		}
+
+		morphoCompound.withdraw(_poolTokenAddress, _amt);
+
+		convertWethToEth(_isETH, TokenInterface(_token), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogWithdraw(address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Payback ETH/ERC20_Token.
+	 * @notice Payback a token to Morpho Compound.
+	 * @param _tokenAddress The address of underlying token to payback.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to payback.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to payback. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens paid back.
+	 */
+	function payback(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		bool _isETH = _tokenAddress == ethAddr;
+		uint256 _amt = getUint(_getId, _amount);
+
+		TokenInterface _tokenContract = _isETH 
+			? TokenInterface(wethAddr) 
+			: TokenInterface(_tokenAddress);
+
+		if (_amt == uint256(-1)) {
+			uint256 _amtDSA = _isETH
+				? address(this).balance
+				: _tokenContract.balanceOf(address(this));
+
+			(, , uint256 _amtDebt) = morphoCompoundLens
+				.getCurrentBorrowBalanceInOf(_poolTokenAddress, address(this));
+
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
+		}
+
+		if (_isETH) convertEthToWeth(_isETH, _tokenContract, _amt);
```

Comment:
```suggestion
		convertEthToWeth(_isETH, _tokenContract, _amt);
```
`_isETH` is checked inside `convertEthToWeth` so it's currently checked twice without intermediary code execution